pages/payment_page.py

```diff
@@ -1,28 +1,32 @@
 from pages.base_page import BasePage
+from selenium.webdriver.common.by import By
 import time
 
 class PaymentPage(BasePage):
+    PAY_TEXTBOX = (By.ID, 'pay-amount')
+    CLICK_OFF_PAY = (By.CLASS_NAME, 'pay-steps')
+    PAY_METHOD = (By.ID,'1119452')
+    AGREE_RADIO = (By.ID, 'agree-terms')
+    SUBMIT_PAYMENT = (By.ID, 'submitPayment')
+
     def __init__(self, driver):
         super().__init__(driver)
+        self.get_clickable_element(self.PAY_TEXTBOX)
 
-        self.locators = {
-            'pay_txt_box': driver.find_element_by_id('pay-amount'),
-            'click_pay': driver.find_element_by_class_name('pay-steps')
-        }
-
-    def update_locators(self):
-        self.locators['payment_method'] = self.wait_for_id('1119452')
-        self.locators['agree_radio'] = self.driver.find_element_by_id('agree-terms')
-        self.locators['submit_payment'] = self.driver.find_element_by_id('submitPayment')
+    
 
     def make_payment(self, amount):
-        self.locators['pay_txt_box'].send_keys(amount)
-        self.locators['click_pay'].click()
-        self.update_locators()
-        self.locators['payment_method'].click()
-        self.click_with_js(self.locators['agree_radio'])
+        pay = self.driver.find_element(*self.PAY_TEXTBOX)
+        pay.send_keys(amount)
+        off_pay = self.driver.find_element(*self.CLICK_OFF_PAY)
+        off_pay.click()
```

Comment:
To cut down on the repeated code, you can make a method called "clickElement" that takes a locator and then searches for and then clicks it. In a flow like this one (make_payment) it will cut down on almost half of the lines. This works too though! No shame in it.